src/auth.js

```diff
@@ -4,14 +4,14 @@ var config = require('./config');
 var models = require('./models');
 var middleware = require('./middleware');
 
-function sendToken (user_id, res, user) {
+function sendToken (user, res) {
   user = user.toJSON();
   user.password = undefined;
-  jwt.sign({}, config.secret, { subject: user_id.toString(), expiresIn: "2h" }, function (err, token) {
+  jwt.sign({}, config.secret, { subject: user.id.toString(), expiresIn: "2h" }, function (err, token) {
     if (err != null) {
       throw err;
     } else {
-      res.send({token: token, userName: user.name});
+      res.send({token: token, user});
```

Comment:
Okay I changed my mind a little. As we just need to send email and name, would be better if keep it:
```js
     res.send({token: token, email: user.email, name: user.password});
```

This way you dont have to worry about hiding values like password from the object. 

Thus you can delete the lines
```js
user = user.toJSON();
user.password = undefined;
```